### ISSUE TITLE
dog food `@remix-run/testing`

### ISSUE THREAD

**Issue by mcansh** [remix-run/remix#5528]
TODO: look into act warnings

```log
console.error
      Warning: An update to RouterProvider inside a test was not wrapped in act(...).
      
      When testing, code that causes React state updates should be wrapped into act(...):
      
      act(() => {
        /* fire events that update state */
      });
      /* assert on the output */
      
      This ensures that you're testing the behavior the user would see in the browser. Learn more at https://reactjs.org/link/wrap-tests-with-act
          at RouterProvider (/remix-run/remix/node_modules/react-router/lib/components.tsx:56:3)
          at RemixStub (/remix-run/remix/packages/remix-testing/dist/create-remix-stub.js:41:5)

```

**Comment by mcansh** [comment]
get baseline of current tests on playwright first.

**Comment by mcansh** [comment]
realizing as im migrating some playwright tests to `@remix-run/testing` that we'd be losing out on _some_ confidence that server renders are behaving as expected due to rendering via react-testing-library along with a few other things we do in our playwright tests

- test assets exist via url (e.g make sure a css file or svg is emitted)
- create arbitrary assets for a test (e.g css files)
- check status codes
- no js experience
- flat routes experience as we're not dealing with a filesystem

gonna migrate the ones i think are safe to do, but we'll more than likely need to keep playwright around for the foreseeable future

**Comment by brophdawg11** [comment]
Closed by #7186
